WordPress/Classes/System/UITestConfigurator.swift

```diff
@@ -1,45 +1,75 @@
-    ///
-    /// XCUIApplication().launchArguments = ["-no-animations"]
-    ///
-    private static func disableAnimations(_ application: UIApplication) {
-        if CommandLine.arguments.contains("-no-animations") {
-            UIView.setAnimationsEnabled(false)
-            application.windows.first?.layer.speed = MAXFLOAT
-            application.mainWindow?.layer.speed = MAXFLOAT
-        }
+    /// It was designed to be used in UI test suites. To enable it just pass a launch argument into XCUIApplicaton.
+    private static func disableAnimations(in app: UIApplication, window: UIWindow) {
+        UIView.setAnimationsEnabled(false)
+        window.layer.speed = MAXFLOAT
     }
 
-    private static func logoutAtLaunch() {
-        if CommandLine.arguments.contains("-logout-at-launch") {
-            removeSelfHostedSites()
-            AccountHelper.logOutDefaultWordPressComAccount()
+    private static func resetEverything() {
+        // Remove CoreData DB
+        ContextManager.shared.resetEverything()
+
+        // Clear user defaults.
+        for key in UserDefaults.standard.dictionaryRepresentation().keys {
+            UserDefaults.standard.removeObject(forKey: key)
         }
     }
 
-    private static func removeSelfHostedSites() {
-        let context = ContextManager.shared.mainContext
-        let service = BlogService(coreDataStack: ContextManager.shared)
-        let blogs = try? BlogQuery().hostedByWPCom(false).blogs(in: context)
-        for blog in blogs ?? [] {
-            service.remove(blog)
-        }
+    private static var flags: UITestFlag = []
+
+    static func isEnabled(_ flag: UITestFlag) -> Bool {
+        flags.contains(flag)
     }
+}
 
-    private static func disableCompliancePopover() {
-        if CommandLine.arguments.contains("-ui-testing") {
-            UserDefaults.standard.didShowCompliancePopup = true
-            UserPersistentStoreFactory.instance().onboardingNotificationsPromptDisplayed = true
-        }
+struct UITestFlag: OptionSet {
+    let rawValue: UInt16
+
+    init(rawValue: UInt16) {
+        self.rawValue = rawValue
     }
+
+    /// Disable all active prompts in the app, such as "Allow Notifications"
+    /// reminders, and others.
+    static let disablePrompts = UITestFlag(rawValue: 1 << 0)
+
+    /// Disable migration from the WordPress app to ensure that it doesn't
+    /// interfere with UI tests on local devices when the app happens to be installed.
+    static let disableMigration = UITestFlag(rawValue: 2 << 0)
+
+    /// Disable password Autofill, preventing the automated prompts from appearing
+    /// during the login to enter any already saved password and after the login
+    /// to save a password.
+    static let disableAutofill = UITestFlag(rawValue: 3 << 0)
+
+    /// Disables all animations, including the ones not managed by `UIView`.
+    static let disableAnimations = UITestFlag(rawValue: 4 << 0)
+
+    static let disableLogging = UITestFlag(rawValue: 5 << 0)
```

Comment:
The raw values above should be `1 << 0`, `1 << 1`, `1 << 2`, etc?

Reply:
I have no idea how I tested it. Thanks.